spec/services/order_syncer_spec.rb

```diff
@@ -34,27 +34,31 @@
           expect(syncer.sync!).to be true
         end
 
-        xit "does not update the shipping_method on the subscription or on the pre-altered shipment" do
+        it "does not update the shipping_method on the subscription or on the pre-altered shipment" do
           expect(order.reload.shipping_method).to eq new_shipping_method
           expect(order.reload.shipments.first.shipping_method).to eq new_shipping_method
           expect(syncer.order_update_issues[order.id]).to be nil
         end
       end
 
       context "when the shipping method on a shipment is not the same as the new shipping method on the subscription" do
-        let(:changed_shipping_method) { create(:shipping_method) }
+        let!(:changed_shipping_method) { create(:shipping_method) }
 
         before do
           # Updating the shipping method on a shipment updates the shipping method on the order,
           # and vice-versa via logic in Spree's shipments controller. So updating both here mimics that
           # behaviour.
-          order.shipments.first.update_attributes(shipping_method_id: changed_shipping_method.id)
-          order.update_attributes(shipping_method_id: changed_shipping_method.id)
+          shipment = order.shipments.first
+          shipping_rate = shipment.shipping_rates.where(shipping_method_id: changed_shipping_method.id).first
+          shipment.update_attributes(selected_shipping_rate_id: shipping_rate.id)
```

Comment:
this is probably order.shipment.select_shipping_method(changed_shipping_method.id)